src/Microsoft.Data.Analysis/DataFrameRowCollection.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Microsoft.Data.Analysis
+{
+    /// <summary>
+    /// Represents the rows of a <see cref="DataFrame"/>
+    /// </summary>
+    public class DataFrameRowCollection : IEnumerable<DataFrameRow>
+    {
+        private readonly DataFrame _dataFrame;
+
+        /// <summary>
+        /// Initializes a <see cref="DataFrameRowCollection"/>.
+        /// </summary>
+        internal DataFrameRowCollection(DataFrame dataFrame)
+        {
+            _dataFrame = dataFrame ?? throw new ArgumentNullException(nameof(dataFrame));
+        }
+
+        /// <summary>
+        /// An indexer to return the <see cref="DataFrameRow"/> at <paramref name="index"/>
+        /// </summary>
+        /// <param name="index">The row index</param>
+        public DataFrameRow this[long index]
+        {
+            get
+            {
+                return new DataFrameRow(_dataFrame, index);
+            }
+        }
+
+        /// <summary>
+        /// Returns an enumerator of <see cref="DataFrameRow"/> objects
+        /// </summary>
+        public IEnumerator<DataFrameRow> GetEnumerator()
+        {
+            for (long i = 0; i < Count; i++)
+            {
+                yield return new DataFrameRow(_dataFrame, i);
+            }
+        }
+
+        /// <summary>
+        /// The number of rows in this <see cref="DataFrame"/>.
+        /// </summary>
+        public long Count => _dataFrame.Columns.RowCount;
```

Comment:
It might be weird that `DataFrameColumnCollection` stores `RowCount`. It started off that way because I needed to know how many rows are in the `DataFrame` and `DataFrameColumnCollection` didn't have a reference to `DataFrame`. We can move `RowCount` to `DataFrameRowCollection`, but it doesn't give us any benefit yet, so I'd vote to keep it this way for now until we have reason to do the refactoring work.

Reply:
I agree is is weird that DataFrameColumnCollection stores RowCount. However, I don't see a benefit in moving it at this point. So I'm fine the way it is, for now.